### PR TITLE
Migrate DesignWare I2C driver to new PCIe system

### LINKED ISSUE #13963: up_squared: evaluate removal of SBL-related special configurations
PCI enumeration seems to be a requirement for determining MMIO addresses on this platform; see #13928. 
As such the special configuration files for SBL-boot (vs standard BIOS/UEFI) are probably unnecessary and can be eliminated to prevent bitrot.

### PR DESCRIPTION
**Pull request by ghost**
This only affects the galileo and up_squared boards, as they are the only targets that use the PCI functionality of the driver.  I've refactored a few things to make adding future Apollo Lake targets easier.

Fixes: #13963

### CODE REVIEW

**Comment by gnuless** [comment]
I have physically confirmed the driver still works properly on a galileo with an I2C blinkM.

File: boards/x86/galileo/Kconfig.defconfig [hunk 1]
```diff
@@ -40,7 +40,7 @@ config GPIO_PCAL9535A_0_DEV_NAME
 config GPIO_PCAL9535A_0_I2C_ADDR
 	default 0x25
 config GPIO_PCAL9535A_0_I2C_MASTER_DEV_NAME
-	default "$(dt_str_val,DT_SNPS_DESIGNWARE_I2C_90007000_LABEL)"
+	default "$(dt_str_val,DT_SNPS_DESIGNWARE_I2C_0_LABEL)"
```
Comment (andyross) [inline boards/x86/galileo/Kconfig.defconfig]: You can reference DTS values in kconfig?  I had no idea.

**Review by andyross — APPROVED** [review]
Honestly the devicetree stuff is a little beyond me, but the C changes look great.

File: drivers/i2c/CMakeLists.txt [hunk 2]
```diff
@@ -33,6 +32,19 @@ zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2
 	i2c_ll_stm32.c
 	)
 
+if(CONFIG_I2C_DW)
+  zephyr_library_sources(i2c_dw.c)
+  foreach(NUM RANGE 0 7)
+    if(CONFIG_I2C_${NUM})
```
Comment (andrewboie) [inline drivers/i2c/CMakeLists.txt]: This is apparently making the Kconfig checks unhappy, @ulfalizer any ideas?

Comment (gnuless) [reply drivers/i2c/CMakeLists.txt]: I'll put in a PR to the ci-tools to clear up the warning.

Comment (ulfalizer) [reply drivers/i2c/CMakeLists.txt]: Yeah, might have to whitelist that one for now.

If more cases like this show up, then maybe the token pasting false positive fix that @Sizurka added could be wrangled into detecting this case too.

Comment (ulfalizer) [reply drivers/i2c/CMakeLists.txt]: Maybe something like `r"\bCONFIG_[A-Z0-9_]+\b(?!$|\s*##)"` would work (see https://github.com/zephyrproject-rtos/ci-tools/commit/05b727a477612f27d8ba45edde3e76f80ae4da3d), adding `$` to negative lookahead.

Comment (ulfalizer) [reply drivers/i2c/CMakeLists.txt]: Going through the regex right now and whitelisting some other stuff as well (`CONFIG_$VAR$` and `CONFIG_FOO_*`, last one for comments).

If `ci-tools` was in the same repo, I could tweak things in the same PR, but no such luck...

Comment (ulfalizer) [reply drivers/i2c/CMakeLists.txt]: PR: https://github.com/zephyrproject-rtos/ci-tools/pull/65

Comment (gnuless) [reply drivers/i2c/CMakeLists.txt]: Thanks @ulfalizer .. I really didn't want to mess with that regex.

File: boards/x86/galileo/Kconfig.defconfig (hunk 1 above)
Comment (ulfalizer) [reply boards/x86/galileo/Kconfig.defconfig]: Was added by @galak. See `scripts/kconfig/kconfigfunctions.py`.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by gnuless** [comment]
Force-push to restart CI checks after @ulfalizer's updates.

**Comment by ulfalizer** [comment]
recheck

**Comment by gnuless** [comment]
> Looks like `recheck` is broken.

Yeah. I've force-pushed again.. fingers crossed!

**Pull request merged**